x/ccv/child/keeper/distribution.go

```diff
@@ -0,0 +1,69 @@
+package keeper
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	clienttypes "github.com/cosmos/ibc-go/modules/core/02-client/types"
+	channeltypes "github.com/cosmos/ibc-go/modules/core/04-channel/types"
+	"github.com/cosmos/interchain-security/x/ccv/child/types"
+	// XXX delete or uncomment this block before merge
+	//ibctypes "github.com/cosmos/ibc-go/v3/modules/apps/transfer/types"
+)
+
+// Simple model, donate tokens to the fee pool of the provider validator set
+// reference: cosmos/ibc-go/modules/apps/transfer/keeper/msg_server.go
+func (k Keeper) DonateToProviderValidatorSet(ctx sdk.Context) error {
+	if !k.shouldTransmit(ctx) {
+		return nil
+	}
+
+	port := k.GetDistributionTransmissionPort(ctx)
+	channel := k.GetDistributionTransmissionChannel(ctx)
+
+	// work around to reuse the IBC token transfer logic
+	consumerFeePoolAddr := k.accountKeeper.GetModuleAccount(ctx, k.feeCollectorName).GetAddress()
+	providerFeePoolAddrStr := k.GetProviderFeePoolAddrStr(ctx)
+
+	tokens := k.bankKeeper.GetAllBalances(ctx, consumerFeePoolAddr)
+	for _, token := range tokens {
+
+		// XXX delete or uncomment this block before merge
+		//if !strings.HasPrefix(token.Denom, "ibc/") {
+		//    denomTrace := ibctypes.ParseDenomTrace(token.Denom)
+		//    token.Denom = denomTrace.IBCDenom()
+		//}
```

Comment:
@AdityaSripal any ideas if I need this block that I've left commented out here?

Reply:
Ahh yes you will need this if you expect that people will pay fees on the consumer chain with tokens they got from somewhere else.  this is because the internal representation of IBC coins in bank is different than the on-the-wire representation of the coin in IBC packets.

See: https://github.com/cosmos/ibc-go/blob/main/docs/architecture/adr-001-coin-source-tracing.md

Reply:
sweet. Thanks for the clarification. uncommented this block.